src/plugins/controls/public/options_list/components/options_list_popover_suggestions.tsx

```diff
@@ -41,6 +47,26 @@ export const OptionsListPopoverSuggestions = ({
   const existsSelected = optionsList.select((state) => state.explicitInput.existsSelected);
   const selectedOptions = optionsList.select((state) => state.explicitInput.selectedOptions);
 
+  const dataViewId = optionsList.select((state) => state.output.dataViewId);
+
+  const [fieldFormatter, setFieldFormatter] = useState(() => (toFormat: string) => toFormat);
+
+  /**
+   * derive field formatter from fieldSpec and dataViewId
+   */
+  useMount(() => {
+    (async () => {
+      if (!dataViewId || !fieldSpec) return;
+      // dataViews are cached, and should always be available without having to hit ES.
+      const dataView = await getDataViewById(dataViewId);
+      setFieldFormatter(
+        () =>
+          dataView?.getFormatterForField(fieldSpec).getConverterFor('text') ??
+          ((toFormat: string) => toFormat)
+      );
+    })();
+  });
```

Comment:
Looks like we are doing the exact same thing in `options_list_control.tsx` - can we save this and store it in the component state instead when the control is created? Can this be changed dynamically?

Reply:
Just bumping [this](https://github.com/elastic/kibana/pull/164362#discussion_r1337758684) again 🙈

Reply:
I tried adding it to `componentState` but redux is upset because functions are non-serializable values, so instead of that, I extracted this code into its own hook that both components use now. See [c41a1b1](https://github.com/elastic/kibana/pull/164362/commits/c41a1b1afd189703fb985a07d7346f39f5032a26)